packages/components/atoms/f-segmented-control/src/components/SegmentedControl.vue

```diff
@@ -0,0 +1,254 @@
+                event.preventDefault();
+                this.selectOption(this.options[this.tabIndex].label);
+
+                return;
+            }
+
+            // If the newIndex has changed, update the tabIndex and focus the new button
+            if (newIndex !== this.tabIndex) {
+                this.focusNewButton(newIndex, currentButton);
+            }
+        },
+        focusNewButton (newIndex, currentButton) {
+            this.tabIndex = newIndex;
+            // Remove focus from the current button
+            currentButton.blur();
+
+            // Wait for the DOM to update, then focus the new button
+            this.$nextTick(() => {
+                this.$el.children[this.tabIndex].focus();
+            });
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use '@justeat/fozzie/src/scss/fozzie' as f;
+
+// Variables
+$sc-sm-height: f.spacing(f);
+$sc-l-height: calc(f.spacing(b) * 6);
+$sc-border-radius: 9999px;
+$sc-padding-xs-modifier: calc(f.spacing(a) / 2);
+$sc-background-clr: f.$color-container-strong;
+$sc-background-clr-selected: f.$color-container-default;
+
+// Segmented control container
+.c-segmented-control {
+    display: flex;
+    justify-content: space-between;
+    background-color: $sc-background-clr;
+    border-radius: $sc-border-radius;
+    gap: 2.5px; // Ensure outlines and box-shadows don't overlap
+    padding: $sc-padding-xs-modifier;
+    min-block-size: $sc-sm-height; // Small size by default
+    user-select: none;
+
+    // Large size modifier
+    &--large {
+        min-block-size: $sc-l-height;
+
+        .c-segmented-control__option {
+            padding-block: calc($sc-padding-xs-modifier + f.spacing(b));
+        }
+    }
+}
+
+// Segmented control options
+.c-segmented-control__option {
+    font-size: inherit;
+    color: inherit;
+    flex: 1; // Distribute available space equally among buttons
+    border-radius: inherit;
+    border: none;
+    padding: f.spacing(a) f.spacing(e);
+    font-weight: f.$font-weight-regular;
+    background: none;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    gap: f.spacing(b);
+    outline: none; // Replace outline with box-shadow on focus due to iOS Safari not respecting border-radius
+
+    // Selected state
+    &--selected {
+        font-weight: f.$font-weight-bold;
+        background-color: f.$color-container-default;
+        box-shadow: f.$elevation-02;
+
+        // Hover state
+        &:hover:not(:disabled) {
+            // Prevent hover effects from being applied on touch devices
+            @media(hover: hover) and (pointer: fine) {
+                background-color: darken($sc-background-clr-selected, f.$color-hover-01);
+            }
+        }
+
+        // Active state
+        &:active:not(:disabled) {
+            background-color: darken($sc-background-clr-selected, f.$color-active-01);
+        }
+    }
+
+    // Not disabled state
```

Comment:
I would put all these descendent styles above the `--selected {` styling, to group all the base component state styling together